src/Reopt/TypeInference/Constraints.hs

```diff
@@ -549,9 +549,12 @@ trivialSubC ctx s t = subtype' resolveVar resolveVar s t
 -- | @absurdSubC s t ctx@ returns @True@ if `s <: t` is an absurd constraint given `ctx`.
 -- N.B., this is intended to be a conservative check (i.e., sound but incomplete).
 absurdSubC ::  Context -> Ty -> Ty -> Bool
-absurdSubC ctx s t = not $ subtype' resolveL resolveR s t
-  where resolveL x = findLowerBound x (ctxVarBoundsMap ctx)
-        resolveR x = findUpperBound x (ctxVarBoundsMap ctx)
+absurdSubC ctx s t = not $ subtype' resolveLower resolveUpper s t
+  where resolveVar findBound x = case Map.lookup x $ ctxVarEqMap ctx of
+                                   Nothing -> findBound x (ctxVarBoundsMap ctx)
+                                   Just xTy -> xTy
```

Comment:
I would replace this with:
```
fromMaybe
  (findBound x (ctxVarBoundsMap ctx))
  (Map.lookup x (ctxVarEqMap ctx))
```

Reply:
If this is just preference... I prefer the way I wrote it =)

Reply:
Happy to tweak it later or have it tweaked elsewhere if you feel that's truly best. I'm going to merge now so the fix is in `main`.